File: src/main/java/lightninglobby/lightninglobby/Lobbynodamage.java
package lightninglobby.lightninglobby;

import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;

public class Lobbynodamage implements Listener {
    @EventHandler

    public void onPlayerDamage(EntityDamageEvent e){
        if (e.getEntity() instanceof Player && e.getEntity().getWorld().getName().equals("Minigames_Lobby")){
            e.setCancelled(true);
        }
        return;
    }
}

File: src/main/kotlin/lightninglobby.lightninglobby/Lobbynodamage.kt
package lightninglobby.lightninglobby

import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent

class Lobbynodamage : Listener {
    @EventHandler
    fun onPlayerDamage(e: EntityDamageEvent) {
        if (e.entity is Player && e.entity.world.name == "Minigames_Lobby") {
            e.isCancelled = true
        }
        return
    }
}